openwisp_controller/config/base/whois.py

```diff
@@ -0,0 +1,59 @@
+from django.db import models
+from django.utils.translation import gettext_lazy as _
+from swapper import get_model_name
+
+from openwisp_utils.base import TimeStampedEditableModel
+
+
+class AbstractWHOISInfo(TimeStampedEditableModel):
+    """
+    Abstract model to store WHOIS information
+    for a device.
+    """
+
+    device = models.OneToOneField(
+        get_model_name('config', 'Device'),
+        on_delete=models.CASCADE,
+        related_name='whois_info',
+        help_text=_('Device to which this WHOIS info belongs'),
+    )
+    last_public_ip = models.GenericIPAddressField(
+        db_index=True,
+        help_text=_(
+            'indicates the IP address logged from '
+            'the last request coming from the device'
+        ),
```

Comment:
I think we can get rid of this help text as we won't show this in the admin I believe as it's going to be redundant with `last_ip`.

Reply:
Let's add basic validation which raises a ValidationError if someone tries to save this with a private IP.